### PR TITLE
circumflex: init at 2.6

### PR DESCRIPTION
**Pull request by mktip**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/circumflex/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    repo = "circumflex";
+    owner = "bensadeh";
+    rev = "2.6";
+    sha256 = "sha256-pcY2PXiOazKAi8mAAbmftXDae01fcUw/u9JPOHQVclI=";
+  };
+
+  subPackage=".";
+  vendorHash = "sha256-rF1Hu4Pf9AF2MTx4GAPmzSn0M38uTxPS1bsAkO23SdI=";
+
+  buildInputs = [ less ];
```
Comment (K900) [inline pkgs/applications/networking/circumflex/default.nix:17]: Does this really need to be in `buildInputs`?

File: pkgs/applications/networking/circumflex/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  version = "2.6";
+
+  src = fetchFromGitHub {
+    repo = "circumflex";
+    owner = "bensadeh";
+    rev = "2.6";
+    sha256 = "sha256-pcY2PXiOazKAi8mAAbmftXDae01fcUw/u9JPOHQVclI=";
+  };
+
+  subPackage=".";
```
Comment (K900) [inline pkgs/applications/networking/circumflex/default.nix:14]: Pretty sure this should be OK to remove.

File: pkgs/applications/networking/circumflex/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ lib, less, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "circumflex";
+  version = "2.6";
+
+  src = fetchFromGitHub {
+    repo = "circumflex";
+    owner = "bensadeh";
+    rev = "2.6";
```
Comment (K900) [inline pkgs/applications/networking/circumflex/default.nix:10]: Maybe use `version` here just to have one place to update later?

File: pkgs/applications/networking/circumflex/default.nix (hunk 1 above)
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix:17]: If I understood it correctly buildInputs are the runtime requirements in a sense? If not then I'll remove it, if yes then I found the following in the application I'm packaging (they seem to use `less` behind the scenes)

https://github.com/bensadeh/circumflex/blob/4af60b5e88c61ebd265bcb2353a61aac8900afbf/cli/cli.go#L32

File: pkgs/applications/networking/circumflex/default.nix (hunk 2 above)
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix:14]: You are right! I removed it and everything still ran fine. I would assume its the default value?

File: pkgs/applications/networking/circumflex/default.nix (hunk 3 above)
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix:10]: can you elaborate further? You mean version as an argument to `fetchFromGithub`? what should it contain? and where can I read more about it if possible?

File: pkgs/applications/networking/circumflex/default.nix (hunk 1 above)
Comment (K900) [reply pkgs/applications/networking/circumflex/default.nix:17]: Not quite. `buildInputs` will be available in the build sandbox, but are not automatically made available at runtime - anything used at runtime needs to refer to the full store path. It looks like the application expects to find `less` in PATH (and also `clear`, which is part of `ncurses`), so it probably needs to be wrapped using [`wrapProgram`](https://nixos.org/manual/nixpkgs/unstable/#fun-makeWrapper).

File: pkgs/applications/networking/circumflex/default.nix (hunk 3 above)
Comment (K900) [reply pkgs/applications/networking/circumflex/default.nix:10]: I mean using `rev = version;`, so you get the `version` value from the outer attrset.

File: pkgs/applications/networking/circumflex/default.nix (hunk 2 above)
Comment (K900) [reply pkgs/applications/networking/circumflex/default.nix]: Yes.

**Review by 06kellyjac — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/circumflex/default.nix (hunk 1 above)
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix]: hmm I see. I fixed it as you suggested, I used `wrapProgram` with `less` and `ncurses`.

File: pkgs/applications/networking/circumflex/default.nix (hunk 3 above)
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix]: I see, makes sense. I changed it as requested.

File: pkgs/applications/networking/circumflex/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{ lib, less, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "circumflex";
+  version = "2.6";
+
+  src = fetchFromGitHub {
+    repo = "circumflex";
+    owner = "bensadeh";
```
[the remark this answers is not in the document]
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix]: Hmm I see, so always have the owner name first, then follow with the repo name. Fixed it and will keep it in mind the next time I submit a package

[the remark this answers is not in the document]
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix]: btw is there any specific ordering preferred as well for the inputs of the whole expression? (`buildGoModule`, `lib`, and the rest)

File: pkgs/applications/networking/circumflex/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    rev = "2.6";
```
[the remark this answers is not in the document]
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix]: changed it as requested.

File: pkgs/applications/networking/circumflex/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    sha256 = "sha256-pcY2PXiOazKAi8mAAbmftXDae01fcUw/u9JPOHQVclI=";
+  };
+
+  vendorHash = "sha256-rF1Hu4Pf9AF2MTx4GAPmzSn0M38uTxPS1bsAkO23SdI=";
+
+  buildInputs = [ less ];
+
+  meta = with lib; {
+    description = "a command line tool for browsing Hacker News in your terminal";
```
[the remark this answers is not in the document]
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix]: Capitalized as requested

File: pkgs/applications/networking/circumflex/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+  postInstall = ''
+    wrapProgram $out/bin/clx \
+      --prefix PATH : ${lib.makeBinPath [ less ncurses ]}
+  '';
+
+  meta = with lib; {
+    description = "A command line tool for browsing Hacker News in your terminal";
+    homepage = "https://github.com/bensadeh/circumflex";
+    license = licenses.agpl3;
```
Comment (06kellyjac) [inline pkgs/applications/networking/circumflex/default.nix:26]:
```suggestion
    license = licenses.agpl3;
    mainProgram = "clx";
```

File: pkgs/applications/networking/circumflex/default.nix [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+buildGoModule rec {
+  pname = "circumflex";
+  version = "2.6";
+
+  src = fetchFromGitHub {
+    owner = "bensadeh";
+    repo = "circumflex";
+    rev = version;
+    sha256 = "sha256-pcY2PXiOazKAi8mAAbmftXDae01fcUw/u9JPOHQVclI=";
```
Comment (06kellyjac) [inline pkgs/applications/networking/circumflex/default.nix:11]:
```suggestion
    hash = "sha256-pcY2PXiOazKAi8mAAbmftXDae01fcUw/u9JPOHQVclI=";
```

if using `vendorHash` can also call this `hash` to be consistent (IIRC)

**Review by 06kellyjac — COMMENTED** [review]
mostly looks good

IDK if I would wrap in `less` though since it probably picks up `$PAGER`?
Most systems have less or a `$PAGER` set already

**Comment by mktip** [comment]
> mostly looks good
> 
> IDK if I would wrap in less though since it probably picks up $PAGER?
> Most systems have less or a $PAGER set already

Hmm, the thing is the original application is directly calling `less` and `clear`, but I would see why its not really needed given most systems already have both. Shall I remove the wrap?
 
(original source use of `less`): https://github.com/bensadeh/circumflex/blob/4af60b5e88c61ebd265bcb2353a61aac8900afbf/cli/cli.go#L32

File: pkgs/applications/networking/circumflex/default.nix [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  postInstall = ''
+    wrapProgram $out/bin/clx \
+      --prefix PATH : ${lib.makeBinPath [ less ncurses ]}
+  '';
+
+  meta = with lib; {
+    description = "A command line tool for browsing Hacker News in your terminal";
+    homepage = "https://github.com/bensadeh/circumflex";
+    license = licenses.agpl3;
+    mainProgram = "clx";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/circumflex/default.nix:27]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also squash all the package related commits.

File: pkgs/applications/networking/circumflex/default.nix [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+  vendorHash = "sha256-rF1Hu4Pf9AF2MTx4GAPmzSn0M38uTxPS1bsAkO23SdI=";
+
+  buildInputs = [ less ];
+
+  meta = with lib; {
+    description = "a command line tool for browsing Hacker News in your terminal";
+    homepage = "https://github.com/bensadeh/circumflex";
+    license = licenses.agpl3;
+    platforms = platforms.linux;
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/networking/circumflex/default.nix]: The checkboxes do nothing. ofborg automatically builds it for the 4 most common platforms.

File: pkgs/applications/networking/circumflex/default.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/networking/circumflex/default.nix]: The inputs are mostly first lib and then the rest alphabetical but that is not that strict.

File: pkgs/applications/networking/circumflex/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/applications/networking/circumflex/default.nix]: This is a strong requirement and should be done everywhere possible.

File: pkgs/applications/networking/circumflex/default.nix (hunk 9 above)
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix]: Added myself as a maintainer and pushed the new commit. Let me know if anything else is required.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/networking/circumflex/default.nix (hunk 9 above)
Comment (mktip) [reply pkgs/applications/networking/circumflex/default.nix:28]: I apologize, I removed `mainProgram = "clx";` by mistake. I just re-added it.

**Pull request merged**